src/typeahead/typeahead.spec.ts

```diff
@@ -0,0 +1,211 @@
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['+one', 'one more']);
+           expect(getNativeInput(compiled).value).toBe('o');
+
+           getDebugInput(fixture.debugElement).triggerEventHandler('keyup.enter', {});
+           fixture.detectChanges();
+           expect(getWindow(compiled)).toBeNull();
+           expect(getNativeInput(compiled).value).toBe('one');
+           expect(fixture.componentInstance.model).toBe('one');
+         });
+       })));
+
+    it('should select previous/next results with up/down arrow keys', async(inject([TestComponentBuilder], (tcb) => {
+         const html = `<input type="text" [(ngModel)]="model" [ngbTypeahead]="find"/>`;
+
+         tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
+           fixture.detectChanges();
+           const compiled = fixture.nativeElement;
+
+           fixture.componentInstance.model = 'o';
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['+one', 'one more']);
+
+           // down
+           getDebugInput(fixture.debugElement).triggerEventHandler('keyup.arrowDown', {});
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['one', '+one more']);
+
+           getDebugInput(fixture.debugElement).triggerEventHandler('keyup.arrowDown', {});
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['+one', 'one more']);
+
+           // up
+           getDebugInput(fixture.debugElement).triggerEventHandler('keyup.arrowUp', {});
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['one', '+one more']);
+
+           getDebugInput(fixture.debugElement).triggerEventHandler('keyup.arrowUp', {});
+           fixture.detectChanges();
+           expectWindowResults(compiled, ['+one', 'one more']);
+         });
+       })));
+
+  });
+
+  xdescribe('conversion', () => {
+
+    it('should use objects', async(inject([TestComponentBuilder], (tcb) => {
+         const html = `<input type="text" [(ngModel)]="model" [ngbTypeahead]="findObjects"/>`;
+
+         tcb.overrideTemplate(TestComponent, html).createAsync(TestComponent).then((fixture) => {
+           fixture.detectChanges();
+           const compiled = fixture.nativeElement;
+
+           // TODO
+         });
+       })));
+
+  });
+
+});
+
+@Component(
+    {selector: 'test-cmp', directives: [NGB_TYPEAHEAD_DIRECTIVES], precompile: [NgbTypeaheadWindow], template: ''})
+class TestComponent {
+  model = '';
+
+  find = (text$: Observable<string>) => {
+    const values = ['one', 'one more', 'two', 'three'];
+    return text$.map(text => values.filter(v => v.startsWith(text)));
+  };
+
+  findNothing = (text$: Observable<string>) => { return text$.map(text => []); };
+
+  findObjects = (text$: Observable<string>) => {
+
+    const values = [{id: 1, value: 'one'}, {id: 10, value: 'one more'}, {id: 2, value: 'two'}, {id: 3, value: 'three'}];
```

Comment:
Maybe move this is as an instance of a component? This is what would be happening in "real life", no?